internal/pkg/cli/env_init.go

```diff
@@ -67,17 +78,33 @@ type initEnvOpts struct {
 	profileConfig profileNames
 	prog          progress
 
-	// initialize profile-specific env clients
-	initProfileClients func(*initEnvOpts) error
+	// Initialize clients after Ask().
+	configureRuntimeClients func(*initEnvOpts) error
 }
 
-var initEnvProfileClients = func(o *initEnvOpts) error {
-	profileSess, err := sessions.NewProvider().FromProfile(o.EnvProfile)
-	if err != nil {
-		return fmt.Errorf("create session from profile %s: %w", o.EnvProfile, err)
+func configureInitEnvClients(o *initEnvOpts) error {
+	var sess *session.Session
+	if o.EnvProfile != "" {
+		profileSess, err := sessions.NewProvider().FromProfile(o.EnvProfile)
+		if err != nil {
+			return fmt.Errorf("create session from profile %s: %w", o.EnvProfile, err)
+		}
+		sess = profileSess
+	} else {
+		staticSess, err := session.NewSession(&aws.Config{
+			Credentials: credentials.NewStaticCredentials(o.TempCreds.AccessKeyID, o.TempCreds.SecretAccessKey, o.TempCreds.SessionToken),
```

Comment:
Do we need to nil check `o.TempCreds`?

Reply:
One of `TempCreds` or `EnvProfile` will be initialized after the `Ask()` method but I haven't implemented that in this PR 😋

Right now this PR doesn't surface these flags yet to the customer, the behavior is exactly the same as before it justs enables the new codepath during `Execute`.

Reply:
Ooohh. I think that was what I was missing -- I was going to mention above in the description of the options fields that referring to the `EnvProfile` as `AWS Profile` might be confusing but that I assumed the flag description would use the actual flag name.

Reply:
Added comments to "Profile" as well saying that it's mutually exclusive with "TempCreds"